lib/checks/keyboard/has-at-least-one-main.js

```diff
@@ -0,0 +1,3 @@
+var main = document.querySelector('main,[role=main]');
```

Comment:
What purpose does adding 'virtualnode' serve? I am getting errors concerning an undefined object.

Reply:
The change to use "const main = axe.utils.querySelectorAll(virtualNode, 'main,[role=main')" seems to break all our test cases and we get the following error:

"TypeError: Cannot read property 'length' of undefined"

Is there something we need to change in the test cases?

Reply:
@jongund When you call the check in a test, and that test uses virtualNode, you need to pass a virtualNode object in. We've got a test utility to help set that up on `axe.testUtils.checkSetup`. You can see an example here:

https://github.com/dequelabs/axe-core/blob/develop/test/checks/aria/required-children.js#L22

Reply:
The test utilities are also documented here: https://github.com/dequelabs/axe-core/blob/master/doc/developer-guide.md#test-utilities

Reply:
It doesn't appear to be possible to test the rule on an 'html' node using checkSetup. Do you have any suggestions on how to go about doing this?

Reply:
Forgive me for trying to catch up on this PR...but why do you need to test the HTML node for a `main` rule? Tests are scoped to the `#fixture` element, which is already present in the test environment, to avoid auditing the test harness UI itself.

Reply:
The selector of the rule itself is 'html' so that the rule runs even if there are no main landmarks. My thought was that it wouldn't really be a proper test if it didn't begin from the specified node.